src/serac/numerics/mesh_utils.cpp

```diff
@@ -269,94 +269,135 @@ std::shared_ptr<mfem::ParMesh> buildCylinderMesh(int radial_refinement, int elem
   return extruded_pmesh;
 }
 
-std::shared_ptr<mfem::ParMesh> buildHollowCylinderMesh(int radial_refinement, int elements_lengthwise,
-                                                       double inner_radius, double outer_radius, double height,
-                                                       const MPI_Comm comm)
+std::shared_ptr<mfem::Mesh> buildRing(int radial_refinement, 
+				      double inner_radius, double outer_radius, 
+				      double total_angle, int sectors)
 {
-  static constexpr int dim                   = 2;
-  static constexpr int num_vertices          = 16;
-  static constexpr int num_elems             = 8;
-  static constexpr int num_boundary_elements = 16;
+  using index_type = int;
+  using size_type  = std::vector<index_type>::size_type;
+
+  static constexpr int dim = 2;
+
+  SLIC_ASSERT_MSG(total_angle > 0., "only positive angles supported");
+
+  // ensure total_angle is (0, 2 * pi]
+  total_angle        = std::min(total_angle, 2. * M_PI);
+  const double angle = total_angle / sectors;
+
+  auto num_elems             = static_cast<size_type>(sectors);
+  auto num_vertices_ring     = static_cast<size_type>((total_angle == 2. * M_PI) ? sectors : sectors + 1);
+  auto num_vertices          = num_vertices_ring * 2;
+  auto num_boundary_elements = num_elems * 2;
 
   SLIC_ERROR_IF(outer_radius <= inner_radius,
                 "Outer radius is smaller than inner radius while building a cylinder mesh.");
 
-  double vertices[num_vertices][dim];
-  for (int i = 0; i < 8; i++) {
-    double s       = sin(M_PI_4 * i);
-    double c       = cos(M_PI_4 * i);
+  std::vector<std::vector<double>> vertices(static_cast<size_type>(num_vertices), std::vector<double>(dim, 0.));
```

Comment:
@joshessman-llnl Any way out of this grossness? I know there's a fundamental mismatch between mfem indices and size types.

Reply:
I'd have to mess around with it to be sure but doing some of the calculations as `size_t` might help, e.g., starting with changing the `sectors` parameter to `size_t` and going from there.